src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java

```diff
@@ -223,6 +238,41 @@ public class XdocsPagesTest {
     private static final Set<String> GOOGLE_MODULES = Collections.unmodifiableSet(
         CheckUtil.getConfigGoogleStyleModules());
 
+    private static final String XDOCS_ENCODING = "UTF-8";
+    private static final String XDOCS_HINT = "xdocs-template";
+
+    /**
+     * We want all xdocs to be generated from templates before validation.
+     *
+     * @throws Exception if something goes wrong
+     */
+    @Test
+    @Order(1)
+    public void parseAllTemplates() throws Exception {
```

Comment:
Method name is not intuitive; isn't this where we are actually generating content? Maybe call it `generateXDoccontent`?

 Also, can we just do:
1. Make this method static
2. Use `@BeforeAll` instead of `@Order`
3. Remove `@Test`

This way, have have no need for suppressions for "no assertion in test method". Also, BeforeAll ensures this method is always ran before the others, even when running one of the other test methods by itself, to help avoid weird issues for contributors. `BeforeAll` makes it clear this is a preparatory step, not an actual test.

Reply:
No order should be used. Just make single test method that does all.

Reply:
@romani we can’t run tests on xdoc content until it is generated :)

Reply:
Done. Used `@BeforeAll`, renamed to `generateXdocContent` and made it `static`.

Reply:
Leaving this unresolved since @romani left a comment on this. I will let him resolve it.

Reply:
Befoall is ok for now